src/components/CommentForm.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState, ChangeEvent, FormEvent } from 'react';
+
+type CommentFormProps = {
+  handleSubmit: (text: string, parentId?: string | null) => void;
+  submitLabel: string;
+  hasCancelButton?: boolean;
+  handleCancel?: () => void;
+  initialText?: string;
+};
+
+const CommentForm: React.FC<CommentFormProps> = function ({
+  handleSubmit,
+  submitLabel,
+  hasCancelButton = false,
+  handleCancel,
+  initialText = '',
+}) {
+  const [text, setText] = useState(initialText);
+  const isTextareaDisabled = text.length === 0;
```

Comment:
I didn't get this. why are you disabling it when the length is zero?

Reply:
i actually dont need it i forgot to deleted it sorry for that i just removed this unnecessary code

Reply:
the comment button will not submit anything if the there isnt any comment 
you should write something in the comment so you can submit it , so if the length is zero the button will be disabled .. 
do you think this is a bad approach for it ?

Reply:
no this is a very good approch but the naming is misleading here, it means the the textarea is disabled and not the submit button
also you don't have to put this in a variable, you can use it directly in the button

```jsx
<Button
  disabled={text.length === 0}
  ...
/>
```

Reply:
yes thats right , i did it without a variable direct implementation as you suggested